src/github/client/github-app-client.ts

```diff
@@ -25,11 +25,10 @@ export class GitHubAppClient extends GitHubClient {
 	constructor(
 		logger?: Logger,
 		baseUrl?: string,
-		appId = envVars.APP_ID
+		appId = envVars.APP_ID,
+		privateKey = PrivateKey.findPrivateKey() || ""
```

Comment:
You don't need to pass this in, you can call the defaultKeyLocator function with the `appId`

Reply:
This is still a problem.  We can't have key logic in multiple places, it has to be in a single place for consistency.

`appId` isn't unique, but what would be needed for the keylocator to make it so?  Why aren't we using that as a parameter here?

Reply:
The `appId` is different from `GitHubAppId` and is used by keyLocator to create JWT. This is used as issuer field in JWT.